File: C_Scripting/portpeek.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/socket.h>
#include <netdb.h>
#include <arpa/inet.h>
#include <netinet/in.h>
#include <unistd.h>

int main(int argc, char *argv[]){

	if (argc<3) {
	fprintf(stdout, "Usage: %s <IP> <Port>\n", argv[0]);
	return 1;
	}

	int sockfd;
	int conn_result;
	struct sockaddr_in target;

	sockfd = socket(AF_INET,SOCK_STREAM,0);
	if (sockfd < 0) {
		perror("socket");
			return 1;
	}

	target.sin_family = AF_INET;
	target.sin_port = htons(atoi(argv[2]));
	target.sin_addr.s_addr = inet_addr(argv[1]);

	conn_result = connect(sockfd, (struct sockaddr *)&target, sizeof target);

	if (conn_result == 0){
		fprintf(stdout, "Port %s is open.\n", argv[2]);
	} else {
		fprintf(stdout, "Port %s is closed.\n", argv[2]);
	}

	close(sockfd);
	return 0;
}

File: Python_Scripting/portpeek.py
#!/usr/bin/python3
import socket
import sys

if len(sys.argv) != 3:
	print(f"Usage: {sys.argv[0]} <IP> <Port>")
	sys.exit(1)

ip = sys.argv[1]
port = int(sys.argv[2])

mysocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
resp = mysocket.connect_ex((ip,port))

if (resp == 0):
	print(f"Port {port} is open.")
else:
	print(f"Port {port} is closed.")